### ISSUE TITLE
Update graphic

### ISSUE THREAD

**Issue by casperdcl** [tqdm/tqdm#155]
The front-page `gif` should:
- [ ] use a more accurate "sleep" which doesn't make it falsely appear that `tqdm` is slowing things down when in actual fact it's the `sleep` function to blame
- [ ] include manual usage example
- [ ] include hook examples with e.g. pandas
- [ ] include CLI examples
- [ ] not be so lossy
  - (I compressed it and then @lrq3000 deleted frames manually, rather than the other way around)
- [ ] possibly be broken down into smaller animations inserted at relevant points in the readme
  - to not look too overwhelming
  - to make it easier to add future animations without having to recompile the whole main graphic

**Comment by lrq3000** [comment]
We can use LICEcap for the screen capture in GIF directly, it's working
great and the final file size is usually small, but we can always optimize
manually after to reduce further.

2016-04-23 13:49 GMT+02:00 Casper da Costa-Luis <notifications@github.com>:

**Comment by grantjenks** [comment]
What's the editor that's doing auto-complete in the gif? It looks really useful but it's also a little distracting as docs wiz by.

**Comment by casperdcl** [comment]
It's bpython (I think the latest version of bpython doesn't work though due to silly reworking of ncurses etc). My intention was more to show that (lots) of docs do exist; didn't expect anyone to try and read it ;)

**Comment by lrq3000** [comment]
I made gifs for the IPython module, I just used LICEcap, and the sizes were between 40KB and 75KB, so it seems we don't even need to optimize further.

I also moved all gifs in an `images` folder to be cleaner.

**Comment by CrazyPython** [comment]
use gifv

**Comment by CrazyPython** [comment]
@casperdcl Gifv is a gif format that's much smaller and thus more frames and less lossy.

**Comment by casperdcl** [comment]
Created an mp4 screen capture (1.7MB), then extracted high quality png and removed ends and duplicate frames (13.4MB). Left with this:

[gif_raw.zip](https://github.com/tqdm/tqdm/files/562646/gif_raw.zip)

I've tried combining into a gif but can't get better than 11MB or so. Seems to me that much better than that should be possible considering the source is a 1.7MB mp4.

**Comment by CrazyPython** [comment]
@casperdcl gifv is essentially a compatible soundless mp4. Much better compression, both size and quality.

**Comment by CrazyPython** [comment]
Ooh, a `tqdm contributor` tag! (automatic, yes, but cool)

**Comment by lrq3000** [comment]
Ah nice @CrazyPython , I don't know if we can set the contributors ourselves, but that's a nice Github feature.

To optimize gifs automatically, there is ezgif.com/optimize which uses opensource softwares, if it doesn't help then probably nothing else will (apart from manually merging redundant frames).

Also to screencapture directly into a gif, there is [LICEcap](https://github.com/justinfrankel/licecap), an opensource software too. It works very well with high quality animation for a small size, I used it a lot. Maybe it also provide methods to convert from MP4 to gif or optimization methods on already made gifs but I'm not sure.

**Comment by casperdcl** [comment]
licecap doesn't work on linux (I think) and the mp4 was lossless audioless. i was just interested in mp4 to gif (or png/jpg to gif) tools which compress well semi-lossless. I believe gifv actually tries to give browsers mp4 and webm instead of gif since they are vastly superior formats in terms of optimisation features. feel free to have a play with the zip file - i'd be interested in what you can do with it.

**Comment by casperdcl** [comment]
Ok used a bunch of tools and even created a python script to manage it all. Pretty good result:

![tqdm](https://cloud.githubusercontent.com/assets/10780059/19899391/e20b8278-a057-11e6-99a0-3a0941787c81.gif)

**Comment by lrq3000** [comment]
It looks great Casper, good job!
Le 1 Nov. 2016 18:23, "Casper da Costa-Luis" <notifications@github.com> a
écrit :

Ok used a bunch of tools and even created a python script to manage it all.
Pretty good result:

[image: tqdm]
<https://cloud.githubusercontent.com/assets/10780059/19899391/e20b8278-a057-11e6-99a0-3a0941787c81.gif>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/tqdm/tqdm/issues/155#issuecomment-257630897>, or mute
the thread

**Comment by CrazyPython** [comment]
@casperdcl you can embed a mp4 directly into HTML, as a gif:

```
<video preload="auto" autoplay="autoplay" muted="muted" loop="loop" webkit-playsinline="" style="width: 728px; height: 309.956px;">
                <source src="//i.imgur.com/A61SaA1.mp4" type="video/mp4">
            </video>
```

[result](http://imgur.com/A61SaA1)

**Comment by casperdcl** [comment]
merged in #304
